### PR TITLE
PR: gh #25 DS Audio Interface update

### PR DESCRIPTION
**Pull request by krishna-1809**
Following APIs are deprecated and hence removing it from header:

1. dsResetBassEnhancer

2. dsResetSurroundVirtualizer

3. dsResetVolumeLeveller

4. dsResetSurroundVirtualizer

5. dsResetDialogEnhancement

6. dsSetAudioDucking

7. dsIsAudioLoopThru

8. dsEnableLoopThru

9. dsGetAudioDB

10. dsSetAudioDB

11. dsGetAudioMaxDB

12. dsGetAudioMinDB

13. dsGetAudioOptimalLevel

14. dsSetAudioEncoding
 
15. dsGetAudioEncoding

16. dsGetAudioDelayOffset

17. dsSetAudioDelayOffset 

Corrected the Header description for few APIs with regards to source and sink

### CODE REVIEW

**Comment by hari22yuva** [comment]
@krishna-1809 Update the description

File: include/dsAudio.h [hunk 1]
```diff
@@ -794,7 +760,8 @@ dsError_t  dsSetGraphicEqualizerMode(intptr_t handle, int mode);
 /**
  * @brief Gets the supported MS12 audio profiles
  *
- * This function will get the list of supported MS12 audio profiles
+ * For sink devices, this function will get the list of supported MS12 audio profiles.
+ * For the source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
Comment (bhanucbp) [inline include/dsAudio.h:764]: in dsAVDTypes.h for audioProfileList variable in structure dsMS12AudioProfileList_t. Add all supported profile string.
remove eg (e.g: "Music,Movie,Voice,Night") add all supported prifles

typedef struct _dsMS12AudioProfileList_t {
    char audioProfileList[MAX_PROFILE_LIST_BUFFER_LEN]; ///< buffer containing the list of comma separated audio profile names for menu list
                                                        ///< **(e.g: "Music,Movie,Voice,Night")**
    int audioProfileCount;                              ///< Total number of profiles
} dsMS12AudioProfileList_t;

**Review by bhanucbp — COMMENTED** [review]
address review comments

File: include/dsAudio.h [hunk 2]
```diff
@@ -211,52 +211,6 @@ dsError_t  dsAudioPortTerm();
  */
 dsError_t  dsGetAudioPort(dsAudioPortType_t type, int index, intptr_t *handle);
```
Comment (bhanucbp) [inline include/dsAudio.h:212]: dsGetAudioFormat - is this function returns the current format of playback stream or the format output to the port?

**Review by bhanucbp — COMMENTED** [review]
address the review comments

File: include/dsAudio.h [hunk 3]
```diff
@@ -211,52 +211,6 @@ dsError_t  dsAudioPortTerm();
...
```
Comment (bhanucbp) [inline include/dsAudio.h:213]: dsAudioOutIsConnected - This is used only for the HEADPHONE port, not for the other ports. Also used only for the sink devices. Update the function description

Comment (bhanucbp) [inline include/dsAudio.h:213]: dsSetStereoMode - function used to set the output port format, update the function description

**Review by bhanucbp — COMMENTED** [review]
address the review comments

File: include/dsAudio.h (hunk 3 above)
Comment (krishna-1809) [reply include/dsAudio.h:213]: For now it is used for Headphones. 
Changing the generic API specific to headphone needs confirmation from Module owners

Comment (krishna-1809) [reply include/dsAudio.h:213]: Its the Output Mode - PCM, Passthrough , etc. Needs to be discussed with Module owners

File: include/dsAudio.h [hunk 4]
```diff
@@ -701,7 +663,8 @@ dsError_t  dsSetSurroundVirtualizer(intptr_t handle, dsSurroundVirtualizer_t vir
 /**
  * @brief Gets the Media Intelligent Steering of the audio port.
  *
- * This function returns enable/disable status of Media Intelligent Steering for the audio port corresponding to specified port handle.
+ * For sink devices, this function returns enable/disable status of Media Intelligent Steering for the audio port corresponding to specified port handle.
+ * For the source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h]: updated

File: include/dsAudio.h [hunk 5]
```diff
@@ -724,7 +687,8 @@ dsError_t  dsGetMISteering(intptr_t handle, bool *enabled);
 /**
  * @brief Set the Media Intelligent Steering of the audio port.
  *
- * This function sets the enable/disable status of Media Intelligent Steering for the audio port corresponding to specified port handle.
+ * For sink devices, this function sets the enable/disable status of Media Intelligent Steering for the audio port corresponding to specified port handle.
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h]: updated

File: include/dsAudio.h [hunk 6]
```diff
@@ -747,7 +711,8 @@ dsError_t  dsSetMISteering(intptr_t handle, bool enabled);
 /**
  * @brief Gets the Graphic Equalizer Mode.
  *
- * This function returns the Graphic Equalizer Mode setting used in the audio port corresponding to the specified port handle.
+ * For sink devices, this function returns the Graphic Equalizer Mode setting used in the audio port corresponding to the specified port handle.
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h]: updated

File: include/dsAudio.h [hunk 7]
```diff
@@ -771,7 +736,8 @@ dsError_t  dsGetGraphicEqualizerMode(intptr_t handle, int *mode);
 /**
  * @brief Sets the Graphic Equalizer Mode.
  *
- * This function sets the Graphic Equalizer Mode setting to be used in the audio port corresponding to the specified port handle.
+ * For sink devices, this function sets the Graphic Equalizer Mode setting to be used in the audio port corresponding to the specified port handle.
+ * For source devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h]: updated

File: include/dsAudio.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h]: updated

Comment (krishna-1809) [reply include/dsAudio.h]: updated

File: include/dsAudio.h [hunk 8]
```diff
@@ -211,52 +211,6 @@ dsError_t  dsAudioPortTerm();
  */
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h:211]: updated

**Review by bhanucbp — APPROVED** [review]
ship it

File: include/dsAudio.h [hunk 9]
```diff
@@ -1122,7 +1053,8 @@ dsError_t  dsGetAudioLevel(intptr_t handle, float *level);
 /**
  * @brief Sets the audio volume level of an audio port.
  * 
- * This function sets the audio volume level to be used on the audio port corresponding to specified port handle.
+ * For sink devices, this function sets the audio volume level to be used on the audio port corresponding to specified port handle.
```
Comment (ddevad) [inline include/dsAudio.h:1056]: Please update properly based on existing product HAL implementation

File: include/dsAudio.h [hunk 10]
```diff
@@ -1142,78 +1074,15 @@ dsError_t  dsGetAudioLevel(intptr_t handle, float *level);
...
- */
-dsError_t  dsGetAudioOptimalLevel(intptr_t handle, float *optimalLevel);
-
 /**
  * @brief Gets the audio delay (in ms) of an audio port
  *
- * This function returns the audio delay (in milliseconds) of audio port with respect to video corresponding to the specified port handle.
+ * For sink devices, this function returns the digital audio delay (in milliseconds) of the digital interfaces(HDMI ARC/eARC, SPDIF).
+ * The Audio delay ranges from 0 to 200 milliseconds.
+ * For source devices, this function returns the digital audio delay (in milliseconds) of the digital interfaces(HDMI)
```
Comment (ddevad) [inline include/dsAudio.h:1082]: Don't we have SPDIF on STBs?

File: include/dsAudio.h [hunk 11]
```diff
@@ -1434,6 +1237,8 @@ dsError_t  dsEnableAudioPort(intptr_t handle, bool enabled);
 /**
  * @brief Enables or Disables MS12 DAPV2 and DE feature
  * 
+ * For source devices,this function returns dsERR_OPERATION_NOT_SUPPORTED always.
+ *
```
Comment (ddevad) [inline include/dsAudio.h:1241]: Can we please check if this API is actually used or not?

File: include/dsAudio.h [hunk 12]
```diff
@@ -1742,9 +1507,9 @@ dsError_t dsAudioAtmosCapsChangeRegisterCB (dsAtmosCapsChangeCB_t cbFun);
 dsError_t dsGetAudioCapabilities(intptr_t handle, int *capabilities);
 
 /**
- * @brief Gets the MS12 capabilities of audio port supported by the platform.
+ * @brief Gets the MS12 capabilities supported by the platform.
  * 
- * This function is used to get the supported MS12 capabilities for the input port supported by the platform.
+ * This function is used to get the supported MS12 capabilities of the platform.
```
Comment (ddevad) [inline include/dsAudio.h:1512]: Do we need to mention that this API is port independent as mentioned in other APIs?

File: include/dsAudio.h [hunk 13]
```diff
@@ -1762,102 +1527,11 @@ dsError_t dsGetAudioCapabilities(intptr_t handle, int *capabilities);
...
- * 
- */
-dsError_t dsResetVolumeLeveller(intptr_t handle);
-
+ 
 /**
  * @brief Enables/Disables associated audio mixing feature.
  *
- * This function will enable/disable associated audio mixing feature of audio port corresponding to specified port handle.
+ * This function will enable/disable associated audio mixing feature of playback stream and it is port independent.
```
Comment (ddevad) [inline include/dsAudio.h:1534]: Can we change it to Playback content

File: include/dsAudio.h [hunk 14]
```diff
@@ -1952,7 +1626,7 @@ dsError_t  dsGetFaderControl(intptr_t handle, int* mixerbalance);
 /**
  * @brief Sets AC4 Primary language
  *
- * This function will set AC4 Primary language of audio port corresponding to specified port handle.
+ * This function will set AC4 Primary language of the playback stream and it is port independent.
```
Comment (ddevad) [inline include/dsAudio.h:1629]: Please change stream to content

**Review by ddevad — CHANGES_REQUESTED** [review]

**Review by AnandPandav — APPROVED** [review]
Looks fine.

File: include/dsAudio.h (hunk 11 above)
Comment (krishna-1809) [reply include/dsAudio.h:1241]: dsEnableMS12Config() is not used in sink and source devices

File: include/dsAudio.h (hunk 10 above)
Comment (krishna-1809) [reply include/dsAudio.h:1082]: SPDIF is supported in one STB

File: include/dsAudio.h [hunk 15]
```diff
@@ -124,12 +124,12 @@ extern "C" {
 typedef void (*dsAudioOutPortConnectCB_t)(dsAudioPortType_t portType, unsigned int uiPortNo, bool isPortCon);
 
 /**
- * @brief Callback function used to notify Audio Format change to the `caller`.
+ * @brief Callback function used to notify Audio Format change of playback stream to the `caller`.
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h:127]: updated

File: include/dsAudio.h [hunk 16]
```diff
@@ -124,12 +124,12 @@ extern "C" {
...
  *
  * HAL Implementation should call this method to deliver updated audio format event
  * to the `caller`.
  * 
- * @param[in] audioFormat : New audio format of the active port. Please refer ::dsAudioFormat_t
+ * @param[in] audioFormat : New audio format of the playback stream. Please refer ::dsAudioFormat_t
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h:132]: updated

File: include/dsAudio.h [hunk 17]
```diff
@@ -211,56 +213,10 @@ dsError_t  dsAudioPortTerm();
...
- * 
- * @see dsGetAudioEncoding()
- */
-dsError_t  dsSetAudioEncoding(intptr_t handle, dsAudioEncoding_t encoding);
-
 /**
  * @brief Gets the current audio format.
  *
- * This function returns the current audio format of the specified audio output port(like PCM, DOLBY AC3). Please refer ::dsAudioFormat_t
+ * This function returns the audio format of the current playback stream(like PCM, DOLBY AC3 etc.) and it is port independent. Please refer ::dsAudioFormat_t
```
[the remark this answers is not in the document]
Comment (krishna-1809) [reply include/dsAudio.h:219]: updated

File: include/dsAudio.h (hunk 9 above)
Comment (krishna-1809) [reply include/dsAudio.h:1056]: Please find the updated description below:

**For sink devices, this function sets the audio volume level to be used for Speaker(dsAUDIOPORT_TYPE_SPEAKER) and Headphone(dsAUDIOPORT_TYPE_HEADPHONE) ports.**

File: include/dsAudio.h (hunk 12 above)
Comment (krishna-1809) [reply include/dsAudio.h:1512]: updated

File: include/dsAudio.h (hunk 14 above)
Comment (krishna-1809) [reply include/dsAudio.h:1629]: Done

File: include/dsAudio.h (hunk 11 above)
Comment (ddevad) [reply include/dsAudio.h:1243]: Ok if we are putting a note that the API is deprecated.  Will remove it after internal review.

**Review by ddevad — APPROVED** [review]

File: include/dsAudio.h (hunk 10 above)
Comment (srinivasgtl) [reply include/dsAudio.h]: Do we need to have this open? If the question is clarified, let's resolve.

File: include/dsAudio.h (hunk 15 above)
[the remark this answers is not in the document]
Comment (ddevad) [reply include/dsAudio.h]: Resolved

File: include/dsAudio.h (hunk 16 above)
[the remark this answers is not in the document]
Comment (ddevad) [reply include/dsAudio.h]: Resolved

File: include/dsAudio.h (hunk 10 above)
Comment (ddevad) [reply include/dsAudio.h]: Resolved

File: include/dsAudio.h (hunk 14 above)
Comment (ddevad) [reply include/dsAudio.h]: Resolved

Comment (ddevad) [reply include/dsAudio.h]: Resolved

File: include/dsAudio.h (hunk 13 above)
Comment (ddevad) [reply include/dsAudio.h]: Resolved

File: include/dsAudio.h (hunk 9 above)
Comment (ddevad) [reply include/dsAudio.h]: Looks ok

File: include/dsAudio.h (hunk 17 above)
[the remark this answers is not in the document]
Comment (ddevad) [reply include/dsAudio.h]: Resolved

**Review by ddevad — APPROVED** [review]

**Review by ddevadas53276 — APPROVED** [review]

**Review by bhanucbp — APPROVED** [review]
ship it

**Pull request merged**